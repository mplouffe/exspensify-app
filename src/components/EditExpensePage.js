import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.editExpense}
                />
                <button
                    onClick={this.removeExpense}
                >
                    Remove
                </button>
            </div>
        );
    };
    editExpense = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    removeExpense = () => {
        this.props.removeExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };
}

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
});

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);