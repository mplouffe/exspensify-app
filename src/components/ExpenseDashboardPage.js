import React from 'react';

const ExpenseDashboardPage = () => (
    <div>
        This is from my dashboard component In the New File.
    </div>
);

export default ExpenseDashboardPage;