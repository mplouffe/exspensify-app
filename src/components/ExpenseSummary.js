import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div>
        {props.expenses.length === 1 ? (
            <p>Viewing 1 expense totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}</p>
        ) : (
            <p>Viewing {props.expenses.length} expenses totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}</p>
        )}
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseSummary);