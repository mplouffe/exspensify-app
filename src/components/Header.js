import { NavLink } from 'react-router-dom';
import React from 'react';


const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Main</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    </header>
);

export default Header;