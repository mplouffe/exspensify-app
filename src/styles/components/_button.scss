.button {
    background: $light-blue;
    border: none;
    border-bottom: .2rem solid darken($light-blue, 10%);
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
    border-bottom: none;
}

.button--secondary {
    background: $light-grey;
    border-bottom: .2rem solid darken($light-grey, 10%);
}