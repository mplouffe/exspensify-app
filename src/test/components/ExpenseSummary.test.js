import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render Expense summary with a single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={195} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render Expense summary with a multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={114195} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with no expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={0} expensesTotal={0} />);
    expect(wrapper).toMatchSnapshot();
});