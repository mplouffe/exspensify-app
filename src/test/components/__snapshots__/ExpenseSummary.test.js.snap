// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense summary with a multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </h1>
</div>
`;

exports[`should render Expense summary with a single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
